packages/xo-server/src/xapi/mixins/storage.mjs

```diff
@@ -52,18 +52,22 @@ export default {
     await this._unplugPbd(this.getObject(id))
   },
 
-  _getUnhealthyVdiChainLength(uuid, childrenMap, cache) {
+  _getUnhealthyVdiChainLength(uuid, childrenMap, cache, vdisWithUnknownVhdParent) {
     let length = cache[uuid]
     if (length === undefined) {
       const children = childrenMap[uuid]
       length = children !== undefined && children.length === 1 ? 1 : 0
       try {
         const parent = this.getObjectByUuid(uuid).sm_config['vhd-parent']
         if (parent !== undefined) {
-          length += this._getUnhealthyVdiChainLength(parent, childrenMap, cache)
+          length += this._getUnhealthyVdiChainLength(parent, childrenMap, cache, vdisWithUnknownVhdParent)
         }
       } catch (error) {
-        log.warn(`Xapi#_getUnhealthyVdiChainLength(${uuid})`, { error })
+        if (error instanceof Error && error.message.startsWith('no object with UUID:')) {
```

Comment:
I'm not found of these kind of tests, usually I like error messages to be changeable in case we want to make it clearer.

You can pass a default value as a second param to `getObjectByUuid()` (e.g. `undefined`) and test this case instead.